src/Microsoft.Azure.SignalR.AspNet/Infrastructure/ServiceConnectionContainer.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.SignalR.Protocol;
+
+namespace Microsoft.Azure.SignalR.AspNet
+{
+    internal class ServiceConnectionContainer : IServiceConnectionContainer
+    {
+        private readonly IReadOnlyList<IServiceConnection> _serviceConnections;
+
+        public ServiceConnectionContainer(Func<IServiceConnection> generator, int count)
+        {
+            if (generator == null)
+            {
+                throw new ArgumentNullException(nameof(generator));
+            }
+
+            if (count <= 0)
+            {
+                throw new ArgumentException($"{nameof(count)} must be greater than 0.");
+            }
+
+            _serviceConnections = Enumerable.Repeat(generator(), count).ToList();
+        }
+
+        public Task StartAsync()
+        {
+            return Task.WhenAll(_serviceConnections.Select(c => c.StartAsync()));
+        }
+
+        public Task StopAsync()
+        {
+            return Task.WhenAll(_serviceConnections.Select(c => c.StopAsync()));
+        }
+
+        public Task WriteAsync(ServiceMessage serviceMessage)
+        {
+            var index = StaticRandom.Next(_serviceConnections.Count);
+            return _serviceConnections[index].WriteAsync(serviceMessage);
+        }
+
+        public async Task WriteAsync(string partitionKey, ServiceMessage serviceMessage)
+        {
+            if (_serviceConnections.Count == 0) return;
+
+            // If we hit this check, it is a code bug.
+            if (string.IsNullOrEmpty(partitionKey))
+            {
+                throw new ArgumentNullException(nameof(partitionKey));
+            }
+
+            var index = Math.Abs(partitionKey.GetHashCode() % _serviceConnections.Count);
```

Comment:
My prefer: `(partitionKey.GetHashCode() & int.MaxValue) % _serviceConnections.Count`

Reply:
cc @xscript How about changing the partition algorithm as we are not yet releasing the SDK?

Reply:
Sure. I am okay with it.

Reply:
Will update it in another pr